configure

```diff
@@ -0,0 +1,375 @@
+        CXX=*|CXX_FOR_BUILD=*|CXXFLAGS_FOR_BUILD=*|CPPFLAGS_FOR_BUILD=*\
+        |LDFLAGS_FOR_BUILD=*|CXXFLAGS=*|CXXFLAGS_EXTRA=*|TEST_CXXFLAGS=*\
+        |TEST_CXXFLAGS_EXTRA|LDFLAGS=*|LDFLAGS_EXTRA=*|TEST_LDFLAGS=*\
+        |TEST_LDFLAGS_EXTRA=*|CPPFLAGS=*) eval "${arg%%=*}=\${arg#*=}" ;;
+        *=*) warning Unknown variable: "${arg%%=*}" ;;
+        *) warning Unknown argument: "$arg" ;;
+    esac
+done
+
+## General Defines
+: "${PLATFORM:=$(uname)}"
+: "${PREFIX:="/usr"}"
+: "${EPREFIX:="/"}"
+: "${NO_SANITIZE:="auto"}"
+: "${SBINDIR:="/sbin"}"
+: "${MANDIR:="/share/man/"}"
+: "${NO_SANITIZE:="auto"}"
+: "${SHUTDOWN_PREFIX:=""}"
+: "${CXXFLAGS_EXTRA:=""}"
+: "${TEST_CXXFLAGS_EXTRA:=""}"
+: "${CPPFLAGS:=""}"
+: "${LDFLAGS_EXTRA:=""}"
+: "${TEST_LDFLAGS_EXTRA:=""}"
+if [ "$PLATFORM" = "Linux" ]; then
+    : "${BUILD_SHUTDOWN:="yes"}"
+    : "${SUPPORT_CGROUPS:="1"}"
+    : "${SYSCONTROLSOCKET:="/run/dinitctl"}"
+else
+    : "${BUILD_SHUTDOWN:="no"}"
+    : "${SUPPORT_CGROUPS:="0"}"
+    : "${SYSCONTROLSOCKET:="/var/run/dinitctl"}"
+fi
+
+## Resolve final paths
+SBINDIR="$EPREFIX/$SBINDIR"
+MANDIR="$PREFIX/$MANDIR"
+
+## Finalize $CXXFLAGS, $TEST_CXXFLAGS, $LDFLAGS, $TEST_LDFLAGS
+if [ -z "${CXXFLAGS+IS_SET}" ]; then
+    CXXFLAGS=""
+    AUTO_CXXFLAGS="true"
+else
+    AUTO_CXXFLAGS="false"
+fi
+if [ -z "${TEST_CXXFLAGS+IS_SET}" ]; then
+    TEST_CXXFLAGS=""
+    AUTO_TEST_CXXFLAGS="true"
+else
+    AUTO_TEST_CXXFLAGS="false"
+fi
+if [ -z "${LDFLAGS+IS_SET}" ]; then
+    LDFLAGS=""
+    AUTO_LDFLAGS="true"
+else
+    AUTO_LDFLAGS="false"
+fi
+if [ -z "${TEST_LDFLAGS+IS_SET}" ]; then
+    TEST_LDFLAGS=""
+    AUTO_TEST_LDFLAGS="true"
+else
+    AUTO_TEST_LDFLAGS="false"
+fi
+
+## Verify PLATFORM value
+if [ "$PLATFORM" != "Linux" ] && [ "$PLATFORM" != "FreeBSD" ] && \
+[ "$PLATFORM" != "OpenBSD" ] && [ "$PLATFORM" != "Darwin" ]; then
+    warning "$PLATFORM" platform is unknown!
+    sub_warning Known Platforms are: Linux, FreeBSD, OpenBSD, Darwin
+fi
+
+## Create testfile.cc to test c++ compiler
+echo "int main(int argc, char **argv) { return 0; }" > testfile.cc || error Cant create test file
+
+## Find and test C++ compiler
+if [ -z "${CXX:-}" ]; then
+    info Checking C++ compiler...
+    for guess in g++ clang++ c++; do
+        findcxx "$guess"
+        if [ -n "${CXX:-}" ]; then
+            sub_info "$CXX"
+            break # Found
+        fi
+    done
+    if [ -z "${CXX:-}" ]; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+cxx_works "$CXX"
+if [ -n "${CXX_FOR_BUILD:-}" ]; then
+    cxx_works "$CXX_FOR_BUILD"
+fi
+
+## Test compiler/linker supported arguments
+if [ "$AUTO_CXXFLAGS" = "true" ]; then
+    for argument in -std=c++11 \
+                    -Wall \
+                    -Os \
+                    -fno-plt
+    do
+        try_cxx_argument CXXFLAGS $argument
+    done
+    if [ "$PLATFORM" != "FreeBSD" ]; then
+        try_cxx_argument CXXFLAGS -flto
```

Comment:
`-flto` needs to be passed to linker as well as compiler, and in that case, _all_ the optimisation options also need to be passed to the compiler.

Reply:
(As a consequence: you probably should only try `-flto` if both `AUTO_CXXFLAGS` and `AUTO_LDFLAGS` are true).

Also: there's no harm in trying to use LTO on FreeBSD, since if it fails you will not apply the flags anyway. That's the whole point of testing them :) (The `-fno-rtti` case is different because it won't cause compilation failure, but causes failure at runtime).